### ISSUE TITLE
VPN fails to initiate handshake

### ISSUE THREAD

**Issue by JKingZapata** [mozilla-mobile/mozilla-vpn-client#2693]
**Description**

After updating to 2.7, users are unable to connect to the VPN, when tapping the switch to connect it is stuck in "attempting to confirm connection" until it switches back to off position. From the VPN logs that have been shared, it keeps trying to initiate handshake 
```
[04.02.2022 17:12:46.473] Info: (windows - tunnel.dll) [TUN] [MozillaVPN] peer(c2if?MrBE) - Retrying handshake because we stopped hearing back after 15 seconds
[04.02.2022 17:12:46.473] Info: (windows - tunnel.dll) [TUN] [MozillaVPN] peer(c2if?MrBE) - Sending handshake initiation
```

**Technical Information**

Platform: Windows 10/11
FX product: Mozilla VPN 2.7

**Steps to Reproduce**
1. Purchase a VPN subscription
2. Install VPN on Windows
3. Flip switch to turn On VPN

**Actual Results**
- When tapping the switch to connect it is stuck on connecting status, then it goes to off

[mozillavpn-2022-2-4 (1).txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/8026316/mozillavpn-2022-2-4.1.txt)

![20220203_114444](https://user-images.githubusercontent.com/72564534/153046587-275edd78-07ed-4e54-afdb-62c55ed9aed4.jpg)

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-1698)

**Comment by JKingZapata** [comment]
attaching another set of VPN logs for Windows 11 
[mozillavpn-2022-2-9.txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/8036467/mozillavpn-2022-2-9.txt)

**Comment by magsmoz** [comment]
It seems like a full reset helps for the moment, but as soon as the device goes into sleep mode and the user turns it back on the VPN won't connect again . I see this more often on Windows 11 than 10 
Here is another set of logs for Win11:
[mozillavpn-2022-2-7_5 (1).txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/8079725/mozillavpn-2022-2-7_5.1.txt)

**Comment by data-sync-user** [comment]
➤ Joseph Anthony Cuevas commented:

regarding this bug
CS had a user giving the following feedback :
"In my case the problem of getting no connection to the VPN occurs when I am connected via ethernet. The WIFI is not disabled and it seems to be interfering with the VPN. Whenever I am connected to ethernet I have to disable WIFI manually to get Mozilla VPN to connect. On the other hand more than one ethernet connections are no problem for the VPN.*To summarize: VPN works when*

  * connected to WIFI only
  * connected to ethernet (one or more connections) with WIFI manually disabled

VPN will not work when

  * connected to ethernet with the WIFI adapter enabled (although the adapter says it is not connected)

I hope this will help you troubleshooting.Some information on my WiFi adapter:
Name: [00000002] Intel(R) Wireless-AC 9560 160MHz
Adapter type: Ethernet 802.3

**Comment by data-sync-user** [comment]
➤ Juan Zapata commented:

Betty Fleming we believe it is critical to have it fixed on 2.8. We have at least 55 confirmed cases, but the number may be much higher and in some cases, our basic troubleshooting is not fixing the issue

**Comment by data-sync-user** [comment]
➤ Betty Fleming commented:

Juan Zapata can you provide any more information on this? We can’t reproduce it… (and please tag Owen Kirby if you have anything 🙂 )

**Comment by data-sync-user** [comment]
➤ Juan Zapata commented:

Betty Fleming We keep getting user reports about this issue. The majority appear to be resolved by uninstalling the VPN, deleting the App data, and reinstalling it again (though it doesn’t appear to be a permanent fix since some users have reached back to us saying they encounter the issue again and have to keep repeating our suggested workaround). We do get some users reporting that the issue happens intermittently or after restarting their device but we haven’t identified a specific condition in which it is always reproducible, so I’m not sure what other information we can collect that would be helpful other than a new set of logs. 

[^mozillavpn-2022-3-22_2-3.txt]

**Comment by data-sync-user** [comment]
➤ Betty Fleming commented:

Owen Kirby Andrea Marchesini

**Comment by data-sync-user** [comment]
➤ Betty Fleming commented:

Valentina Virlics could you see if you can reproduce this?

**Comment by SuperNar3k** [comment]
Today I bought MozillaVPN. Same issue. I am connecting on Windows10Pro. I tried launching MozillaVPN in admin mode, giving it perms to bypass the Windows Firewall. I tried a full reset and uninstall of the application as well. Also restarted the computer a few times and updated all drivers. I tried enabling/disabling Local Network Access, I tried changin the VPN Advanced DNS Settings. Someone above mentioned that when connected to ethernet while wifi is on but not connecting could be the issue; so I turned off wifi such that only ethernet was connecting, and restarted the computer. No change. The VPN simply refuses to connect.

Attached is my log.
[mozillavpn-2022-4-4.txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/8413599/mozillavpn-2022-4-4.txt)

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Betty Fleming We were not able to reproduce this with 2.7 or 2.8.

**Comment by data-sync-user** [comment]
➤ Owen Kirby commented:

At last, I have figured out the steps to reproduce this issue:

  1. Install Mozilla VPN on a Windows 10/11 laptop.
  2. Connect to both Wi-Fi and Ethernet, making sure that:
    1. The Ethernet is gigabit
    2. The Wi-Fi is slower than gigabit

  3. This should produce two default routes, with the Wi-Fi interface having a higher metric. You can ensure this is the case with:
    1. route print from a Windows command prompt, or
    2. Get-Route -AddressFamily IPv4 from Powershell.

  4. Disconnect the Wi-Fi interface but do not disable the interface.
  5. Attempt to start the VPN client.

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.8.0 (2.202204071128), on Windows 11, while following the next steps:

  1. Mozilla VPN was installed, but not turned ON;
  2. While being connected to a Wi-Fi network, the device was also connected to Ethernet;
  3. The two default routes were verified by running the “route print” command in CMD, and the Wi-Fi interface had a higher metric - see the attached screenshot;
  4. The device was disconnected from the Wi-Fi network (but the Wi-Fi was not disabled);
  5. The VPN was successfully turned ON;

Owen Kirby , is there anything else that we should check here? Thank you.

Attaching a video as well. 

!image_2022_04_07T12_36_17_934Z.png|width=1553,height=682!

!fixed.mp4|width=1768,height=1036!

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.8.0 (2.202204071128), on Windows 11, while following the next steps:

  1. Mozilla VPN was installed, but not turned ON;
  2. While being connected to a Wi-Fi network, the device was also connected to Ethernet;
  3. The two default routes were verified by running the “route print” command in CMD, and the Wi-Fi interface had a higher metric - see the attached screenshot;
  4. The device was disconnected from the Wi-Fi network (but the Wi-Fi was not disabled);
  5. The VPN was successfully turned ON;

Owen Kirby , is there anything else that we should check here? Thank you.

Attaching a video as well. 

!image_2022_04_07T12_36_17_934Z.png|width=1553,height=682!

!fixed.mp4|width=1768,height=1036!

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.8.0 (2.202204071128), on Windows 11, while following the next steps:

  1. Mozilla VPN was installed, but not turned ON;
  2. While being connected to a Wi-Fi network, the device was also connected to Ethernet;
  3. The two default routes were verified by running the “route print” command in CMD, and the Wi-Fi interface had a higher metric - see the attached screenshot;
  4. The device was disconnected from the Wi-Fi network (but the Wi-Fi was not disabled);
  5. The VPN was successfully turned ON;

Owen Kirby , is there anything else that we should check here? Thank you.

Attaching a video as well. 

!image_2022_04_07T12_36_17_934Z.png|width=1553,height=682!

!fixed.mp4|width=1768,height=1036!

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.8.0 (2.202204071128), on Windows 11, while following the next steps:

  1. Mozilla VPN was installed, but not turned ON;
  2. While being connected to a Wi-Fi network, the device was also connected to Ethernet;
  3. The two default routes were verified by running the “route print” command in CMD, and the Wi-Fi interface had a higher metric - see the attached screenshot;
  4. The device was disconnected from the Wi-Fi network (but the Wi-Fi was not disabled);
  5. The VPN was successfully turned ON;

Owen Kirby , is there anything else that we should check here? Thank you.

Attaching a video as well. 

!image_2022_04_07T12_36_17_934Z.png|width=1553,height=682!

!fixed.mp4|width=1768,height=1036!

**Comment by pvonahsen** [comment]
I experience the same problem on Win11 and connected only via WiFi
Really frustrating. I was on NordVPN before and it just worked. I trust Mozilla more with my data, but this is unusable :( 
My logfile:
[mozillavpn-2022-7-13_2.txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/9101526/mozillavpn-2022-7-13_2.txt)
